### PR TITLE
Re-enable uglify on the new theme

### PR DESCRIPTION
**Pull request by eternoendless**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | 1.7.3.x
| Description?  | **Problem**: UglifyJs had been disabled at some point because it doesn't work with ES6.<br>**Solution**: transpile ES6 to ES5 using Babel.<br>**Extra magic**: enable tree-shaking<br>**Result**: massive size reduction in javascript file size... 50%~75%
| Type?         | improvement
| Category?     | BO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | n/a
| How to test?  | New pages in the BO should work the same as before (no new javascript errors)

Before:
screen shot 2018-01-08 at 18 37 40
After:
screen shot 2018-01-08 at 18 38 39

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/8659)

### CODE REVIEW

**Comment by Quetzacoalt91** [comment]
To be merged for 1.7.3.0, as it brings back a feature disabled in #8278 for the same milestone.

File: admin-dev/themes/new-theme/webpack.config.js [hunk 1]
```diff
@@ -155,26 +167,28 @@ let config = {
...
+      compress: {
+        sequences: true,
+        conditionals: true,
+        booleans: true,
+        if_return: true,
+        join_vars: true,
+        drop_console: true
+      },
+      output: {
+        comments: false
```
Comment (Quetzacoalt91) [inline admin-dev/themes/new-theme/webpack.config.js]: While we are on that subject, didn't you want to keep the license in the minimized files?

Comment (eternoendless) [reply admin-dev/themes/new-theme/webpack.config.js]: Good point!
I don't want to keep all comments though, I'll try something else

**Comment by eternoendless** [comment]
Done! And with no significative size increase:
screen shot 2018-01-10 at 10 26 40

**Review by Quetzacoalt91 — DISMISSED** [review]

File: admin-dev/themes/new-theme/webpack.config.js [hunk 2]
```diff
@@ -155,26 +168,28 @@ let config = {
...
-//         join_vars: true,
-//         drop_console: true
-//       },
-//       output: {
-//         comments: false
-//       }
-//     })
-//   );
-// } else {
+console.log(process.env.NODE_ENV);
```
Comment (Quetzacoalt91) [inline admin-dev/themes/new-theme/webpack.config.js]: Do we keep that console.log call?

Comment (eternoendless) [reply admin-dev/themes/new-theme/webpack.config.js]: Nope! I forgot that, thanks

**Review by Quetzacoalt91 — APPROVED** [review]

File: admin-dev/themes/new-theme/package.json [hunk 3]
```diff
@@ -73,6 +73,7 @@
     "style-loader": "^0.14.0",
     "tether": "^1.3.2",
     "typeahead.js": "^0.11.1",
+    "uglify-save-license": "^0.4.1",
```
Comment (mickaelandrieu) [inline admin-dev/themes/new-theme/package.json]: is this code removing all licenses from compressed files? This is *something* we don't want isn't it?

**Review by mickaelandrieu — APPROVED** [review]

**Pull request merged**